contrib/gazebo/gazebo_sawyer_env.py

```diff
@@ -0,0 +1,419 @@
+"""
+This is gazebo environment for a specific robot - sawyer.
+Every specific sawyer task environment should inherit it.
+"""
+
+from rllab.envs.base import Step
+from .gazebo_env import GazeboEnv
+from rllab.spaces import Box
+import numpy as np
+import rospy
+import intera_interface
+import copy
+import rospkg
+from rllab.core.serializable import Serializable
+import os.path as osp
+import subprocess
+import os
+import threading
+import time
+from functools import wraps
+
+from gazebo_msgs.msg import (
+    ModelStates,
+)
+
+from geometry_msgs.msg import (
+    Pose,
+    Point
+)
+
+# /robot/joint_states: [j0, j1, j2, j3, j4, j5, j6]
+# /robot/joint_limits: [j0, j1, j2, j3, j4, j5, j6]
+joint_angle_gripper_cmd_lower = [-5.0952, 0.0, -0.020833, -3.0503, -3.8095, -3.0426, -3.0439, -2.9761, -2.9761, -3.14, 0]
+joint_angle_gripper_cmd_upper = [0.9064, 0.020833, 0.0, 3.0503, 2.2736, 3.0426, 3.0439, 2.9761, 2.9761, 3.14, 100]
+
+def goal_distance(goal_a, goal_b):
+    """
+    :param goal_a:
+    :param goal_b:
+    :return: distance between goal_a and goal_b
+    """
+    assert goal_a.shape == goal_b.shape
+    return np.linalg.norm(goal_a - goal_b, axis=-1)
+
+def rate_limited(max_per_second: int):
+    """Rate-limits the decorated function locally, for one process."""
+    lock = threading.Lock()
+    min_interval = 1.0 / max_per_second
+
+    def decorate(func):
+        last_time_called = time.perf_counter()
+
+        @wraps(func)
+        def rate_limited_function(*args, **kwargs):
+            lock.acquire()
+            nonlocal last_time_called
+            try:
+                elapsed = time.perf_counter() - last_time_called
+                left_to_wait = min_interval - elapsed
+                if left_to_wait > 0:
+                    time.sleep(left_to_wait)
+
+                return func(*args, **kwargs)
+            finally:
+                last_time_called = time.perf_counter()
+                lock.release()
+
+        return rate_limited_function
+
+    return decorate
+
+class GazeboSawyerEnv(GazeboEnv, Serializable):
+    """
+    Gazebo Sawyer Env
+    """
+
+    def __init__(self, has_object, distance_threshold,
+                 initial_qpos, reward_type,  target_in_the_air,
+                 limb='right', tip_name="right_gripper_tip", target_range=0.15,
+                 obj_range=0.15):
+        """
+        Input
+        -----
+        control_mode: have 4 different control modes: position, velocity, trajectory-position, torque
```

Comment:
There is no `control_mode` argument. Update the documentation with the current arguments and their meanings.